angel_system/uho/prediction.py

```diff
@@ -0,0 +1,335 @@
+import torch
+from torch.utils.data import DataLoader
+import tqdm
+from typing import List
+from typing import Optional
+from typing import Tuple
+
+from angel_system.utils.simple_timer import SimpleTimer
+from .src.datamodules.ros_datamodule import get_common_transform, AngelDataset
+from .src.models.components.fcn import UnifiedFCNModule
+from .src.models.components.transformer import TemTRANSModule
+from .aux_data import AuxData
+
+import pdb
+
+
+# TODO: Pull in from a resource file instead of defining via hard-coding.
+UHO_LABELS = ["Background",
+              "Measure 12 ounces of water in the liquid measuring cup",
+              "Pour the water from the liquid measuring cup into the electric kettle",
+              "Turn on the Kettle",
+              "Place the dripper on top of the mug",
+              "Take the coffee filter and fold it in half to create a semi-circle",
+              "Fold the filter in half again to create a quarter-circle",
+              "Place the folded filter into the dripper such that the the point of the quarter-circle rests in the center of the dripper",
+              "Spread the filter open to create a cone inside the dripper",
+              "Turn on the kitchen scale",
+              "Place a bowl on the scale",
+              "Zero the scale",
+              "Add coffee beans to the bowl until the scale reads 25 grams",
+              "Pour the measured coffee beans into the coffee grinder",
+              "Set timer for 20 seconds",
+              "Turn on the timer",
+              "Grind the coffee beans by pressing and holding down on the black part of the lid",
+              "Pour the grounded coffee beans into the filter cone prepared in step 2",
+              "Turn on the thermometer",
+              "Place the end of the thermometer into the water",
+              "Set timer to 30 seconds",
+              "Pour a small amount of water over the grounds in order to wet the grounds",
+              "Slowly pour the water over the grounds in a circular motion. Do not overfill beyond the top of the paper filter",
+              "Allow the rest of the water in the dripper to drain",
+              "Remove the dripper from the cup",
+              "Remove the coffee grounds and paper filter from the dripper",
+              "Discard the coffee grounds and paper filter"]
+
+
+# prepare input data
+def collate_fn_pad(batch):
+    """Padds batch of variable length.
+
+    :params batch (Dict): Batch data with labels + auxiliary data from
+        dataloader
+
+    :returns batch (Dict): Processed batch with labels + auxiliary data
+    """
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    aux_data = {}
+
+    lhand, rhand = [], []
+    dets, bbox, frms = [], [], []
+    topK = 10
+    for t in batch:
+        lhand = [torch.from_numpy(tmp).to(device) for tmp in t[0]["l_hand"]]
+        rhand = [torch.from_numpy(tmp).to(device) for tmp in t[0]["l_hand"]]
```

Comment:
These lines look identical, is the second line supposed to be r_hand?

Reply:
Probably. I will correct however this function is not actually utilized in prediction (I think Dawei was using it in testing).